### ISSUE TITLE
Larastan won't run under PHP 8.1 and Laravel 8

### ISSUE THREAD

**Issue by Jamesking56** [nunomaduro/larastan#1217]
- Larastan Version: 2.1.4
- `--level` used: 8
- Pull request with failing test: N/A

### Description

Larastan won't run under PHP 8.1 (Laravel framework version v8.83.9) as it errors when running:

```
> vendor/bin/phpstan analyse --ansi
Note: Using configuration file /var/www/html/phpstan.neon.dist.
PHP Fatal error:  Declaration of Illuminate\Foundation\Application::storagePath() must be compatible with Illuminate\Contracts\Foundation\Application::storagePath($path = '') in /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php on line 458
Fatal error: Declaration of Illuminate\Foundation\Application::storagePath() must be compatible with Illuminate\Contracts\Foundation\Application::storagePath($path = '') in /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php on line 458
Script vendor/bin/phpstan analyse --ansi handling the phpstan event returned with error code 255
```

**Comment by Jamesking56** [comment]
Nevermind, don't run Larastan v2 with Laravel 8

**Comment by florentpoujol** [comment]
Interesting, I run PHPStan from a `devtools` forlder (like PHP-CS-Fixer) and even with Larastan v1.0.3, it will install the Illuminate component v9, by default, since there is no `laravel/framework` entry in the Composer file to restrict their version to 8.x.

So I also got the error mentioned the OP.

To fix, I just added `"laravel/framework": "^8.0",` in PHPStan's folder Composer file so solve the issue.   

Moving PHPStan's installation back to the project's main Composer file is also a solution.
